### PR TITLE
SDI-287 publish prisoner difference events

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerSearchByBookingIdsResourceTest.kt [hunk 1]
```diff
@@ -8,11 +8,13 @@ import org.awaitility.kotlin.matches
 import org.awaitility.kotlin.untilCallTo
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
+import org.springframework.test.context.TestPropertySource
 import org.springframework.web.reactive.function.BodyInserters
 import uk.gov.justice.digital.hmpps.prisonersearch.QueueIntegrationTest
 import uk.gov.justice.digital.hmpps.prisonersearch.services.PrisonerListCriteria.BookingIds
 import uk.gov.justice.digital.hmpps.prisonersearch.services.dto.OffenderBooking
 
+@TestPropertySource(properties = ["index.page-size=12"])
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerSearchByBookingIdsResourceTest.kt:17]: Fixes old bug which caused 4 * number of messages and hence timed out when tests ran locally.

File: helm_deploy/values-dev.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ generic-service:
     INDEX_COMPLETE_THRESHOLD: 500000
     APPLICATIONINSIGHTS_CONFIGURATION_FILE: applicationinsights.dev.json
     DIFF_TELEMETRY: true
+    DIFF_EVENTS: true
```
Comment (mikehalmamoj) [inline helm_deploy/values-dev.yaml:16]: defaults to false in application.yml, currently only turned on in dev.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt [hunk 3]
```diff
@@ -99,22 +100,51 @@ class PrisonerIndexService(
...
+      kotlin.runCatching {
+        getDifferencesByPropertyType(prisoner, storedPrisoner)
+          .takeIf { differences -> differences.isNotEmpty() }
+          ?.also { differences ->
+            domainEventEmitter.emitPrisonerDifferenceEvent(offenderBooking.offenderNo, offenderBooking.bookingNo, differences)
+          }
+      }.onFailure {
+        log.error("prisoner-offender-search.offender.updated event failed with error", it)
+      }
+    }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt:147]: The main change - publish an event if the prisoner has changed.

**Comment by hyderkhanmoj** [comment]
Is creating a prisoner not considered a change?

**Review by hyderkhanmoj — APPROVED** [review]
LGTM

**Comment by mikehalmamoj** [comment]
> Is creating a prisoner not considered a change?

Not yet. I'm thinking of introducing another event for that `prisoner-offender-search.offender.inserted`

**Pull request merged**